# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [1]:
import csv

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            print(f'Column names are {", ".join(row)}')
        else:
            print(f'\t{row[0]}  {row[1]}  {row[2]}  {row[3]}  {row[4]}  {row[5]}  {row[6]}')
        line_count += 1

Column names are firstname, lastname, birthmonth, birthday, birthyear, course, phonenumber
	name1  lastname1  11  26  1998  BS LM  +63 918 302 6210
	name2  lastname2  May  26  2001  Management  927 212 6276
	name3  lastname3  5  16  1997  BS CTM  +63 918 253 7017
	name4  lastname4  Dec  9  97  BS ITE  918 714 4360
	name5  lastname5  Sep  5  97  BS CTM  +63 927 462 1674
	name6  lastname6  March  2  1999  Management  +63 918 629 5268
	name7  lastname7  11  10  98  BS LM  +63 917 369 7388
	name8  lastname8  Aug  28  99  Management Engineering  917 240 6000
	name9  lastname9  July  10  97  Management Engineering  918 226 5748
	name10  lastname10  9  14  2001  CTM  +63 918 342 1717
	name11  lastname11  December  8  98  BS LM  +63 927 708 1308
	name12  lastname12  5  4  97  BS ME  927 747 5497
	name13  lastname13  January  28  1999  BS CTM  +63 917 959 4490
	name14  lastname14  1  11  1999  BS MGT  +63 927 543 4192
	name15  lastname15  6  6  98  Information Technology Entrepreneurship  917 

	name2364  lastname2364  Feb  7  1997  Communications Technology  +63 917 740 5047
	name2365  lastname2365  Oct  15  97  BS ME  917 159 7408
	name2366  lastname2366  12  27  98  Legal Management  +63 918 948 8122
	name2367  lastname2367  July  26  97  LM  +63 918 288 1534
	name2368  lastname2368  5  1  97  Information Technology Entrepreneurship  918 991 8438
	name2369  lastname2369  May  23  97  Communications Technology  +63 917 688 6034
	name2370  lastname2370  1  18  98  BS CTM  +63 917 880 1286
	name2371  lastname2371  April  11  2000  BS LM  917 758 9685
	name2372  lastname2372  7  4  2001  BS LM  +63 927 490 5419
	name2373  lastname2373  5  12  2000  ME  927 776 2687
	name2374  lastname2374  August  27  2000  Information Technology Entrepreneurship  +63 927 429 5762
	name2375  lastname2375  November  2  1999  Management  +63 918 276 8342
	name2376  lastname2376  May  27  99  ITE  918 420 9440
	name2377  lastname2377  9  12  2001  CTM  +63 917 316 7284
	name2378  lastname2378  12

# Names

In [2]:
# Create a LASTNAME LIST

import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    first_name_list = []
    last_name_list = []
    for row in reader:
        last_name_list.append(row['lastname'])
last_name_list

['lastname1',
 'lastname2',
 'lastname3',
 'lastname4',
 'lastname5',
 'lastname6',
 'lastname7',
 'lastname8',
 'lastname9',
 'lastname10',
 'lastname11',
 'lastname12',
 'lastname13',
 'lastname14',
 'lastname15',
 'lastname16',
 'lastname17',
 'lastname18',
 'lastname19',
 'lastname20',
 'lastname21',
 'lastname22',
 'lastname23',
 'lastname24',
 'lastname25',
 'lastname26',
 'lastname27',
 'lastname28',
 'lastname29',
 'lastname30',
 'lastname31',
 'lastname32',
 'lastname33',
 'lastname34',
 'lastname35',
 'lastname36',
 'lastname37',
 'lastname38',
 'lastname39',
 'lastname40',
 'lastname41',
 'lastname42',
 'lastname43',
 'lastname44',
 'lastname45',
 'lastname46',
 'lastname47',
 'lastname48',
 'lastname49',
 'lastname50',
 'lastname51',
 'lastname52',
 'lastname53',
 'lastname54',
 'lastname55',
 'lastname56',
 'lastname57',
 'lastname58',
 'lastname59',
 'lastname60',
 'lastname61',
 'lastname62',
 'lastname63',
 'lastname64',
 'lastname65',
 'lastname66',
 'lastname67',
 'la

# Birthmonth Cleaning

In [3]:
import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    dirty_month_list = []
    for row in reader:
        dirty_month_list.append(row['birthmonth'])
    print(dirty_month_list)

['11', 'May', '5', 'Dec', 'Sep', 'March', '11', 'Aug', 'July', '9', 'December', '5', 'January', '1', '6', '4', '1', 'Apr', '12', '3', '7', '9', '4', 'Oct', '5', '12', 'March', '11', '10', 'May', '4', 'Jul', '12', 'Oct', '7', 'December', '2', '8', 'May', '2', 'Nov', 'September', 'Jan', 'February', 'January', 'Dec', 'August', 'Apr', '10', 'March', '8', '3', 'Dec', '7', 'May', '9', 'May', 'April', 'April', 'August', '9', 'Dec', 'Feb', 'November', 'Mar', 'September', 'Dec', 'November', 'November', 'May', '8', 'December', 'Jun', 'Oct', 'Sep', '2', 'May', '9', 'December', '8', 'Feb', '11', 'Jun', 'February', 'Jan', 'Nov', 'August', 'January', '3', 'Mar', 'Aug', 'Dec', 'Jun', 'November', '9', 'Mar', 'Jan', 'Jan', 'May', '5', 'April', 'August', 'Dec', 'July', 'Mar', '8', 'April', 'May', '8', 'January', '12', 'May', 'May', 'August', 'March', 'Jul', '1', 'Feb', 'Dec', '2', '3', 'March', '4', 'February', 'November', 'June', 'Mar', 'June', '10', 'Jul', 'Feb', '9', 'Apr', 'May', '10', '10', 'June',

In [4]:
# Create a dictionary that assigns the MONTH to its ABBREVIATION

month_number = {'January':'1','February':'2','March':'3','April':'4','May':'5','June':'6',
          'July':'7','August':'8','September':'9','October':'10','November':'11','December':'12'}

month_short = {'January':'Jan','February':'Feb','March':'Mar','April':'Apr','May':'May','June':'Jun',
          'July':'Jul','August':'Aug','September':'Sep','October':'Oct','November':'Nov','December':'Dec'}

clean_month_list = []


# loop through list above to change all abbreviations
for i in dirty_month_list:
    
    # NUMBERS
    if len(i) <= 2:
        for j in month_number:
            if i == month_number[j]:
                clean_month_list.append(j)
                
    # 3 LETTER ABBRV
    elif len(i) == 3:
        for k in month_short:
            if i == month_short[k]:
                clean_month_list.append(k)
                
    else:
        clean_month_list.append(i)
                
clean_month_list        
        

['November',
 'May',
 'May',
 'December',
 'September',
 'March',
 'November',
 'August',
 'July',
 'September',
 'December',
 'May',
 'January',
 'January',
 'June',
 'April',
 'January',
 'April',
 'December',
 'March',
 'July',
 'September',
 'April',
 'October',
 'May',
 'December',
 'March',
 'November',
 'October',
 'May',
 'April',
 'July',
 'December',
 'October',
 'July',
 'December',
 'February',
 'August',
 'May',
 'February',
 'November',
 'September',
 'January',
 'February',
 'January',
 'December',
 'August',
 'April',
 'October',
 'March',
 'August',
 'March',
 'December',
 'July',
 'May',
 'September',
 'May',
 'April',
 'April',
 'August',
 'September',
 'December',
 'February',
 'November',
 'March',
 'September',
 'December',
 'November',
 'November',
 'May',
 'August',
 'December',
 'June',
 'October',
 'September',
 'February',
 'May',
 'September',
 'December',
 'August',
 'February',
 'November',
 'June',
 'February',
 'January',
 'November',
 'August',
 'Januar

In [5]:
month_tally = {}

for i in clean_month_list:
    month_tally.setdefault(i,clean_month_list.count(i))

month_tally

{'November': 277,
 'May': 273,
 'December': 276,
 'September': 232,
 'March': 246,
 'August': 241,
 'July': 249,
 'January': 270,
 'June': 259,
 'April': 267,
 'October': 178,
 'February': 232}

# Birthday Cleaning

In [6]:
import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    clean_day_list = []
    for row in reader:
        clean_day_list.append(row['birthday'])
    print(clean_day_list)

['26', '26', '16', '9', '5', '2', '10', '28', '10', '14', '8', '4', '28', '11', '6', '20', '3', '8', '28', '10', '13', '15', '15', '1', '6', '5', '4', '10', '10', '28', '6', '16', '17', '4', '14', '11', '2', '25', '3', '27', '16', '24', '13', '25', '27', '24', '23', '13', '11', '25', '22', '21', '15', '10', '8', '9', '9', '6', '24', '26', '8', '7', '22', '13', '23', '24', '9', '4', '19', '12', '19', '1', '21', '25', '17', '2', '19', '15', '23', '10', '1', '22', '18', '28', '10', '21', '15', '26', '26', '15', '16', '8', '14', '24', '6', '16', '19', '10', '12', '10', '2', '14', '16', '16', '5', '25', '21', '14', '15', '8', '22', '19', '27', '22', '26', '19', '11', '4', '27', '3', '5', '1', '23', '2', '24', '4', '3', '20', '25', '18', '23', '10', '18', '7', '3', '5', '5', '5', '8', '2', '8', '3', '15', '14', '9', '11', '6', '13', '18', '5', '17', '15', '2', '2', '14', '3', '13', '28', '12', '25', '1', '26', '17', '25', '12', '16', '4', '5', '13', '9', '17', '24', '13', '19', '1', '9', '1'

In [7]:
day_tally = {}

for i in clean_day_list:
    day_tally.setdefault(i,clean_day_list.count(i))

day_tally

{'26': 116,
 '16': 115,
 '9': 98,
 '5': 100,
 '2': 106,
 '10': 111,
 '28': 104,
 '14': 102,
 '8': 111,
 '4': 110,
 '11': 105,
 '6': 99,
 '20': 91,
 '3': 90,
 '13': 119,
 '15': 121,
 '1': 112,
 '17': 116,
 '25': 116,
 '27': 102,
 '24': 108,
 '23': 116,
 '22': 103,
 '21': 123,
 '7': 88,
 '19': 113,
 '12': 98,
 '18': 107}

# Birthyear Cleaning

In [8]:
import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    dirty_year_list = []
    for row in reader:
        dirty_year_list.append(row['birthyear'])
    print(dirty_year_list)

['1998', '2001', '1997', '97', '97', '1999', '98', '99', '97', '2001', '98', '97', '1999', '1999', '98', '99', '97', '1997', '1997', '97', '99', '1999', '97', '97', '97', '1999', '1997', '2000', '1999', '1999', '99', '99', '99', '1999', '2001', '1998', '2000', '2001', '1997', '1998', '99', '98', '1999', '98', '97', '98', '1999', '99', '97', '2001', '98', '1999', '2000', '98', '1998', '1998', '2001', '97', '98', '99', '98', '2000', '2000', '97', '97', '1997', '1997', '2001', '1998', '98', '99', '1999', '1998', '1997', '1998', '1997', '2000', '2000', '2000', '99', '97', '1997', '2000', '1999', '1999', '98', '1998', '97', '1998', '99', '97', '97', '1998', '97', '2001', '1999', '1997', '98', '98', '99', '97', '2001', '2000', '2001', '99', '97', '2001', '99', '1998', '99', '2001', '97', '97', '99', '98', '97', '1997', '1999', '1997', '97', '2000', '2001', '1999', '1999', '98', '2000', '2000', '98', '98', '2000', '1999', '99', '1999', '2000', '1999', '2000', '1998', '1997', '1999', '1999', '

In [9]:
clean_year_list = []
for i in dirty_year_list:
    
    # extract years with 4 DIGITS
    if len(i) == 2:
        clean_year_list.append(i)
    
    # CONCATENATE LAST 2 DIGITS
    else:
        clean_i = i[2]+i[3]
        clean_year_list.append(clean_i)
    
print(clean_year_list)

['98', '01', '97', '97', '97', '99', '98', '99', '97', '01', '98', '97', '99', '99', '98', '99', '97', '97', '97', '97', '99', '99', '97', '97', '97', '99', '97', '00', '99', '99', '99', '99', '99', '99', '01', '98', '00', '01', '97', '98', '99', '98', '99', '98', '97', '98', '99', '99', '97', '01', '98', '99', '00', '98', '98', '98', '01', '97', '98', '99', '98', '00', '00', '97', '97', '97', '97', '01', '98', '98', '99', '99', '98', '97', '98', '97', '00', '00', '00', '99', '97', '97', '00', '99', '99', '98', '98', '97', '98', '99', '97', '97', '98', '97', '01', '99', '97', '98', '98', '99', '97', '01', '00', '01', '99', '97', '01', '99', '98', '99', '01', '97', '97', '99', '98', '97', '97', '99', '97', '97', '00', '01', '99', '99', '98', '00', '00', '98', '98', '00', '99', '99', '99', '00', '99', '00', '98', '97', '99', '99', '98', '99', '99', '98', '01', '98', '99', '00', '97', '97', '98', '99', '99', '99', '99', '98', '97', '99', '98', '00', '00', '00', '99', '97', '98', '99', '00

In [10]:
year_tally = {}

for i in clean_year_list:
    year_tally.setdefault(i,clean_year_list.count(i))

year_tally

{'98': 746, '01': 362, '97': 756, '99': 756, '00': 380}

# Course Cleaning

In [11]:
import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    dirty_course_list = []
    for row in reader:
        dirty_course_list.append(row['course'])
    print(dirty_course_list)


['BS LM', 'Management', 'BS CTM', 'BS ITE', 'BS CTM', 'Management', 'BS LM', 'Management Engineering', 'Management Engineering', 'CTM', 'BS LM', 'BS ME', 'BS CTM', 'BS MGT', 'Information Technology Entrepreneurship', 'BS LM', 'Information Technology Entrepreneurship', 'BS MGT', 'Information Technology Entrepreneurship', 'BS ME', 'LM', 'ITE', 'MGT', 'ME', 'BS ME', 'BS ME', 'CTM', 'BS LM', 'LM', 'Communications Technology', 'BS ITE', 'Management', 'BS ITE', 'BS LM', 'ITE', 'BS MGT', 'MGT', 'Legal Management', 'ME', 'BS LM', 'BS MGT', 'BS LM', 'Management', 'Information Technology Entrepreneurship', 'Management Engineering', 'ME', 'Management', 'Information Technology Entrepreneurship', 'Legal Management', 'BS LM', 'ITE', 'Communications Technology', 'BS MGT', 'BS CTM', 'Management Engineering', 'ME', 'BS CTM', 'BS ME', 'Information Technology Entrepreneurship', 'ITE', 'Communications Technology', 'ITE', 'BS ME', 'Information Technology Entrepreneurship', 'BS ME', 'LM', 'BS CTM', 'MGT', '

In [12]:
# extract COURSE TITLES (no repetition)
unique_names = set(dirty_course_list)

unique_names

{'BS CTM',
 'BS ITE',
 'BS LM',
 'BS ME',
 'BS MGT',
 'CTM',
 'Communications Technology',
 'ITE',
 'Information Technology Entrepreneurship',
 'LM',
 'Legal Management',
 'ME',
 'MGT',
 'Management',
 'Management Engineering'}

In [13]:
[n for n in unique_names]

['Information Technology Entrepreneurship',
 'LM',
 'Management Engineering',
 'MGT',
 'BS ITE',
 'Management',
 'BS LM',
 'BS MGT',
 'ITE',
 'BS ME',
 'CTM',
 'ME',
 'Communications Technology',
 'Legal Management',
 'BS CTM']

In [14]:
unique_names_dictionary = {name: None for name in unique_names}
unique_names_dictionary

{'Information Technology Entrepreneurship': None,
 'LM': None,
 'Management Engineering': None,
 'MGT': None,
 'BS ITE': None,
 'Management': None,
 'BS LM': None,
 'BS MGT': None,
 'ITE': None,
 'BS ME': None,
 'CTM': None,
 'ME': None,
 'Communications Technology': None,
 'Legal Management': None,
 'BS CTM': None}

In [15]:
# ASSIGN A NAME per course variation

for i in unique_names_dictionary:
    final_name = input(i + ": "+"Enter name: ")
    unique_names_dictionary[i] = final_name

Information Technology Entrepreneurship: Enter name: BS IT Entrepreneurship
LM: Enter name: BS Legal Management
Management Engineering: Enter name: BS Management Engineering
MGT: Enter name: BS Management
BS ITE: Enter name: BS IT Entrepreneurship
Management: Enter name: BS Management
BS LM: Enter name: BS Legal Management
BS MGT: Enter name: BS Management
ITE: Enter name: BS IT Entrepreneurship
BS ME: Enter name: BS Management Engineering
CTM: Enter name: BS Comtech
ME: Enter name: BS Management Engineering
Communications Technology: Enter name: BS Comtech
Legal Management: Enter name: BS Legal Management
BS CTM: Enter name: BS Comtech


In [16]:
unique_names_dictionary

{'Information Technology Entrepreneurship': 'BS IT Entrepreneurship',
 'LM': 'BS Legal Management',
 'Management Engineering': 'BS Management Engineering',
 'MGT': 'BS Management',
 'BS ITE': 'BS IT Entrepreneurship',
 'Management': 'BS Management',
 'BS LM': 'BS Legal Management',
 'BS MGT': 'BS Management',
 'ITE': 'BS IT Entrepreneurship',
 'BS ME': 'BS Management Engineering',
 'CTM': 'BS Comtech',
 'ME': 'BS Management Engineering',
 'Communications Technology': 'BS Comtech',
 'Legal Management': 'BS Legal Management',
 'BS CTM': 'BS Comtech'}

In [17]:
import re
clean_course_list = [re.sub(r''.join(course),r''.join(unique_names_dictionary[course]),course) for course in dirty_course_list]


In [18]:
clean_course_list

['BS Legal Management',
 'BS Management',
 'BS Comtech',
 'BS IT Entrepreneurship',
 'BS Comtech',
 'BS Management',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Comtech',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Comtech',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Management Engineering',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Comtech',
 'BS Legal Management',
 'BS Legal Management',
 'BS Comtech',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS Management',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Legal Management',
 'BS Management',
 'BS Legal Management',
 'BS Management',
 'BS IT En

In [19]:
course_tally = {}

for i in clean_course_list:
    course_tally.setdefault(i,clean_course_list.count(i))

course_tally

{'BS Legal Management': 585,
 'BS Management': 629,
 'BS Comtech': 589,
 'BS IT Entrepreneurship': 590,
 'BS Management Engineering': 607}

In [20]:
for i in sorted (course_tally):
    print(i,course_tally[i])

BS Comtech 589
BS IT Entrepreneurship 590
BS Legal Management 585
BS Management 629
BS Management Engineering 607


## Number Cleaning

In [21]:
import csv
with open('recweekdata.csv', newline='') as csvfile:
    reader = csv.DictReader(csvfile)
    dirty_number_list = []
    for row in reader:
        dirty_number_list.append(row['phonenumber'])
    print(dirty_number_list)

['+63 918 302 6210', '927 212 6276', '+63 918 253 7017', '918 714 4360', '+63 927 462 1674', '+63 918 629 5268', '+63 917 369 7388', '917 240 6000', '918 226 5748', '+63 918 342 1717', '+63 927 708 1308', '927 747 5497', '+63 917 959 4490', '+63 927 543 4192', '917 211 9497', '+63 918 185 5162', '927 146 8146', '927 516 4280', '+63 927 958 6909', '+63 917 514 9105', '+63 927 984 7499', '+63 927 699 3694', '917 317 2478', '+63 917 159 3829', '+63 918 898 6972', '917 796 4622', '+63 927 138 7580', '+63 917 195 9244', '+63 917 643 3996', '917 758 3119', '+63 918 669 4265', '+63 917 750 8539', '927 236 9504', '917 438 4447', '+63 917 773 6035', '918 455 4185', '927 889 9252', '927 874 8083', '927 842 3053', '927 923 1234', '+63 918 381 4956', '+63 918 602 9278', '+63 918 532 6748', '917 654 1291', '918 244 1683', '917 818 1190', '+63 917 468 6425', '+63 918 635 1688', '917 821 1896', '927 924 9506', '918 854 2626', '+63 918 438 9745', '917 893 8235', '+63 918 216 5250', '+63 917 257 1057',

In [22]:
clean_number_list = []
for i in dirty_number_list:
    
    # REMOVE SPACES
    i = i.replace(" ","")
    
    # APPEND NUMBERS THAT ARE IN (+63) FORMAT
    if len(i) == 13:
        clean_number_list.append(i)
        
    # CONCATENATE (+63) TO REMAINING NUMBERS
    else:
        new_i = "+63"+i
        clean_number_list.append(new_i)
print(clean_number_list)

['+639183026210', '+639272126276', '+639182537017', '+639187144360', '+639274621674', '+639186295268', '+639173697388', '+639172406000', '+639182265748', '+639183421717', '+639277081308', '+639277475497', '+639179594490', '+639275434192', '+639172119497', '+639181855162', '+639271468146', '+639275164280', '+639279586909', '+639175149105', '+639279847499', '+639276993694', '+639173172478', '+639171593829', '+639188986972', '+639177964622', '+639271387580', '+639171959244', '+639176433996', '+639177583119', '+639186694265', '+639177508539', '+639272369504', '+639174384447', '+639177736035', '+639184554185', '+639278899252', '+639278748083', '+639278423053', '+639279231234', '+639183814956', '+639186029278', '+639185326748', '+639176541291', '+639182441683', '+639178181190', '+639174686425', '+639186351688', '+639178211896', '+639279249506', '+639188542626', '+639184389745', '+639178938235', '+639182165250', '+639172571057', '+639279495098', '+639277666306', '+639188518794', '+63917901286

In [23]:
globe_list = []
smart_list = []
for i in clean_number_list:
    # SMART = 0918
    # test 4th and 5th number
    if i[4] == '1' and i[5] == '8':
        smart_list.append(i)
        
    # GLOBE = everything else
    else:
        globe_list.append(i)
  

In [24]:
globe_count = 0
smart_count = 0
for i in globe_list:
    globe_count += 1
    
for i in smart_list:
    smart_count += 1

number_tally = {'Globe':globe_count,'Smart':smart_count}
number_tally

{'Globe': 2045, 'Smart': 955}

In [25]:
for i in sorted (number_tally):
    print(i,number_tally[i])

Globe 2045
Smart 955


# Writing the new CSV

In [26]:
import csv

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    first_name_list = []
    for row in csv_reader:
        if line_count >= 1:
            first_name_list.append(row[0])
        line_count += 1

In [27]:
rows = zip(first_name_list,last_name_list,clean_month_list,clean_day_list,clean_year_list,clean_course_list,clean_number_list)


In [28]:
import csv

with open('cleanrecweekdata.csv', 'a', newline='') as csvfile:
    order_writer = csv.writer(csvfile, quoting=csv.QUOTE_MINIMAL)
    order_writer.writerow(["firstname","lastname","birthmonth","birthday","birthyear","course","phonenumber"])
    for row in rows:
        order_writer.writerow(row)

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [29]:
from IPython.core.display import HTML

html = """
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{month_tally}</td></tr>
             <tr><td>People born on each day</td><td>{day_tally}</td></tr>
             <tr><td>People born in each year</td><td>{year_tally}</td></tr>
             <tr><td>People coming from each course</td><td>{course_tally}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{number_tally}</td></tr>
         </table>
    </body>
</html>
"""

html_output = html.format(month_tally = month_tally,
                          day_tally = day_tally,
                          year_tally = year_tally,
                          course_tally = course_tally,
                          number_tally = number_tally
                         )

HTML(html_output)

In [30]:
html_css="""
<html>
    <link rel="stylesheet" type="text/css" href="CSS_FILE.css" />
    <div>
        <div align="center"><h1>RecWeek 2019</h1></div>
        <div class="container">
            <div id="bl-main" class="bl-main">
                <section>
                    <div class="bl-box">
                        <h2>
                            same birth month
                        </h2>
                        <div class="bl-content">
                            &nbsp;
                        </div>
                        <div class="divTable Mica">
                            <div class="divTableHeading">       
                                <div class="divTableHead">MONTH</div>
                                <div class="divTableHead">COUNT</div>
                            </div>
                            <div class="divTableBody">
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        January
                                    </div>
                                    <div class="divTableCell">
                                        270
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        February
                                    </div>
                                    <div class="divTableCell">
                                        232
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        March
                                    </div>
                                    <div class="divTableCell">
                                        246
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        April
                                    </div>
                                    <div class="divTableCell">
                                        267
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        May
                                    </div>
                                    <div class="divTableCell">
                                        273
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        June
                                    </div>
                                    <div class="divTableCell">
                                        259
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        July
                                    </div>
                                    <div class="divTableCell">
                                        249
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        August
                                    </div>
                                    <div class="divTableCell">
                                        241
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        September
                                    </div>
                                    <div class="divTableCell">
                                        232
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        October
                                    </div>
                                    <div class="divTableCell">
                                        178
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        November
                                    </div>
                                    <div class="divTableCell">
                                        277
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        December
                                    </div>
                                    <div class="divTableCell">
                                        276
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </section>
                <section>
                    <div class="bl-box">
                        <h2>
                            same birth day
                        </h2>
                        <div class="divTable Mica">
                            <div class="divTableHeading">
                                <div class="divTableHead">DAY</div>
                                <div class="divTableHead">COUNT</div>
                            </div>
                            <div class="divTableBody">
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        1
                                    </div>
                                    <div class="divTableCell">
                                        112
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        2
                                    </div>
                                    <div class="divTableCell">
                                        106
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        3
                                    </div>
                                    <div class="divTableCell">
                                        90
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        4
                                    </div>
                                    <div class="divTableCell">
                                        110
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        5
                                    </div>
                                    <div class="divTableCell">
                                        100
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        6
                                    </div>
                                    <div class="divTableCell">
                                        99
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        7
                                    </div>
                                    <div class="divTableCell">
                                        88
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        8
                                    </div>
                                    <div class="divTableCell">
                                        111
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        9
                                    </div>
                                    <div class="divTableCell">
                                        98
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        10
                                    </div>
                                    <div class="divTableCell">
                                        111
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        11
                                    </div>
                                    <div class="divTableCell">
                                        105
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        12
                                    </div>
                                    <div class="divTableCell">
                                        98
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        13
                                    </div>
                                    <div class="divTableCell">
                                        119
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        14
                                    </div>
                                    <div class="divTableCell">
                                        102
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        15
                                    </div>
                                    <div class="divTableCell">
                                        121
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        16
                                    </div>
                                    <div class="divTableCell">
                                        115
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        17
                                    </div>
                                    <div class="divTableCell">
                                        116
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        18
                                    </div>
                                    <div class="divTableCell">
                                        107
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        19
                                    </div>
                                    <div class="divTableCell">
                                        113
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        20
                                    </div>
                                    <div class="divTableCell">
                                        91
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        21
                                    </div>
                                    <div class="divTableCell">
                                        123
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        22
                                    </div>
                                    <div class="divTableCell">
                                        103
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        24
                                    </div>
                                    <div class="divTableCell">
                                        108
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        25
                                    </div>
                                    <div class="divTableCell">
                                        116
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        26
                                    </div>
                                    <div class="divTableCell">
                                        116
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        27
                                    </div>
                                    <div class="divTableCell">
                                        102
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        28
                                    </div>
                                    <div class="divTableCell">
                                        104
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="bl-content">
                        &nbsp;
                    </div>
                </section>
                <section>
                    <div class="bl-box">
                        <h2>
                            same birth year
                        </h2>
                        <div class="divTable Mica">
                            <div class="divTableHeading">
                                <div class="divTableHead">YEAR</div>
                                <div class="divTableHead">COUNT</div>
                            </div>
                            <div class="divTableBody">
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        1997
                                    </div>
                                    <div class="divTableCell">
                                        756
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        1998
                                    </div>
                                    <div class="divTableCell">
                                        746
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        1999
                                    </div>
                                    <div class="divTableCell">
                                        756
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        2000
                                    </div>
                                    <div class="divTableCell">
                                        380
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        2001
                                    </div>
                                    <div class="divTableCell">
                                        362
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="bl-content">
                        &nbsp;
                    </div>
                </section>
                <section id="bl-work-section">
                    <div class="bl-box">
                        <h2>
                            same course
                        </h2>
                        <div class="divTable Mica">
                            <div class="divTableHeading">
                                <div class="divTableHead">COURSE</div>
                                <div class="divTableHead">COUNT</div>
                            </div>
                            <div class="divTableBody">
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        BS Comtech
                                    </div>
                                    <div class="divTableCell">
                                        589
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        BS IT Entrepreneurship
                                    </div>
                                    <div class="divTableCell">
                                        590
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        BS Legal Management
                                    </div>
                                    <div class="divTableCell">
                                        585
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        BS Management
                                    </div>
                                    <div class="divTableCell">
                                        629
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        BS Management Engineering
                                    </div>
                                    <div class="divTableCell">
                                        607
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="bl-content">
                        <div class="bl-content">
                            &nbsp;
                        </div>
                    </div>
                </section>
                <section id="bl-work-section">
                    <div class="bl-box">
                        <h2>
                            same network
                        </h2>
                        <div class="divTable Mica">
                            <div class="divTableHeading">
                                <div class="divTableHead">NETWORK</div>
                                <div class="divTableHead">COUNT</div>
                            </div>
                            <div class="divTableBody">
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        Globe
                                    </div>
                                    <div class="divTableCell">
                                        2045
                                    </div>
                                </div>
                                <div class="divTableRow">
                                    <div class="divTableCell">
                                        Smart
                                    </div>
                                    <div class="divTableCell">
                                        955
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </section>
            </div>
        </div>
    </div>
</html>
""" 
with open('stats.html',"w+") as file: 
    file.write(html_css)